signature/policy_eval_signedby.go

```diff
@@ -70,6 +71,13 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 			return PolicyRequirementError(fmt.Sprintf("Signature by key %s is not accepted", keyIdentity))
 		},
 		validateSignedDockerReference: func(ref string) error {
+			r := image.Reference().DockerReference()
+			if _, isCanonical := r.(reference.Canonical); isCanonical || reference.IsNameOnly(r) {
+				if !pr.UntaggedReference.matchesDockerReference(image, ref) {
```

Comment:
can we assume people always want to pull-by-digest to match the tag in the signature and just have an "referencesByDigest = prohibit"?